src/helpers/figures/charts/pie_chart.ts

```diff
@@ -216,25 +225,93 @@ function getPieConfiguration(
   config.options.layout = {
     padding: { left: 20, right: 20, top: chart.title ? 10 : 25, bottom: 10 },
   };
-  config.options.plugins!.tooltip!.callbacks!.title = function (tooltipItems) {
-    return tooltipItems[0].dataset.label;
-  };
   config.options.plugins!.tooltip!.callbacks!.label = function (tooltipItem) {
     const { format, locale } = localeFormat;
     const data = tooltipItem.dataset.data;
     const dataIndex = tooltipItem.dataIndex;
     const percentage = calculatePercentage(data, dataIndex);
 
-    const xLabel = tooltipItem.label || tooltipItem.dataset.label;
     const yLabel = tooltipItem.parsed.y ?? tooltipItem.parsed;
     const toolTipFormat = !format && yLabel > 1000 ? "#,##" : format;
     const yLabelStr = formatValue(yLabel, { format: toolTipFormat, locale });
 
-    return xLabel ? `${xLabel}: ${yLabelStr} (${percentage}%)` : `${yLabelStr} (${percentage}%)`;
+    return `${yLabelStr} (${percentage}%)`;
+  };
+  config.options.onHover = onHoverPieChart;
+  //@ts-ignore
+  config.options.plugins!.eventPlugin = {
+    events: ["mouseout"],
   };
+  config.plugins = [
+    ...config.plugins,
+    {
+      id: "eventPlugin",
+      afterEvent(chart, args, _) {
+        if (args.event.type === "mouseout") {
+          unHighlightItems(chart.data.datasets);
+          chart.update();
+          lastHoveredIndex = undefined;
+        }
+      },
+    },
+  ];
   return config;
 }
 
+let lastHoveredIndex: number | undefined = undefined;
+
+function highlightItem(index: number, dataSets) {
+  dataSets.forEach((dataset) => {
+    const backgroundColors = dataset.backgroundColor;
+    if (!backgroundColors) {
+      return;
+    }
+    backgroundColors.forEach((color, i, colors) => {
+      const hexColor = toHex(color.length === 9 ? color.slice(0, -2) : color);
+      colors[i] = i === index ? hexColor : hexColor + "4D";
+    });
+  });
+}
+
+function unHighlightItems(dataSets) {
+  dataSets.forEach((dataset) => {
+    const backgroundColors = dataset.backgroundColor;
+    backgroundColors.forEach((color, i, colors) => {
+      const hexColor = toHex(color.length === 9 ? color.slice(0, -2) : color);
+      colors[i] = hexColor;
+    });
+  });
+}
+
+function onHoverLegend(evt: ChartEvent, item: LegendItem, legend: LegendElement<"pie">) {
+  if (item.index === undefined) {
+    return;
+  }
+  const datasets = legend.chart.data.datasets;
+  highlightItem(item.index, datasets);
+  legend.chart.update();
+}
+
+function onLeaveLegend(evt: ChartEvent, item: LegendItem, legend: LegendElement<"pie">) {
+  const datasets = legend.chart.data.datasets;
+  unHighlightItems(datasets);
+  legend.chart.update();
+}
+
+function onHoverPieChart(evt: ChartEvent, items: ActiveElement[], chart: Chart) {
+  const datasets = chart.data.datasets;
+  if (items[0]) {
+    if (items[0].index !== lastHoveredIndex) {
```

Comment:
using the `lastHoveredIndex` that is global and shared across all charts sounds _very_ suspicious... Is it really necessary ?

Reply:
It's not really necessary, but it will avoid doing useless stuff when we move the mouse on the same pie part. But I'll try to find another way to do it